packages/flame/test/components/component_test.dart

```diff
@@ -200,6 +200,56 @@ void main() {
       );
     });
   });
+
+  group('Component descendants', () {
+    flameGame.test(
+      'length must be equal to the number of added components',
+      (game) async {
+        final component = Component()..add(Component()..add(Component()));
+        await game.ensureAdd(component);
+
+        expect(game.descendants().length, 3);
+      },
+    );
+
+    flameGame.test(
+      'length must be equal to the number of added components including itself',
```

Comment:
Maybe this test name that follows "given-when-then" pattern? Same applies for other test names.
```suggestion
      'length must be equal to the number of added components '
      'when including itself',
```

Reply:
Im embarrassed, but can you show an example? I dont know English well🙈

Reply:
This is very nitpicky, but I like to keep the test name consistent. On the testing limbo there are some conventions people usually go for. For example, the [given-when-then](https://en.wikipedia.org/wiki/Given-When-Then#:~:text=Given%2DWhen%2DThen%20\(GWT,words%20given%2C%20when%20and%20then) pattern, or Arrange Act Assert. This refers to how we order the things inside our tests.

However, personally, I also like it to have a similar logic when naming the tests. So it's usually "x happens when this occurs". I included suggestions below where I show this. Sometimes just modifying "if" for "when" is sufficient.

Just to reiterate, this is nitpicky and is a personal preference.

Reply:
Done. Please check me😊